infra_event_notifier/jira_notifier.py

```diff
@@ -0,0 +1,91 @@
+from typing import Dict, Self
+
+from infra_event_notifier.backends.jira import (
+    JiraConfig,
+    create_or_update_issue,
+)
+from infra_event_notifier.notifier import Notifier
+
+
+class JiraNotifier(Notifier):
+    def __init__(
+        self,
+        jira_api_key: str,
+        jira_url: str,
+        jira_project: str,
+        jira_user_email: str,
+    ) -> None:
+        super().__init__()
+        self.jira_config = JiraConfig(
+            url=jira_url,
+            user_email=jira_user_email,
+            project_key=jira_project,
+            api_key=jira_api_key,
+        )
+
+        # Notification fields
+        self.tags: Dict[str, str] = {}
+        self.issue_type: str = "Task"
+        self.fallback_comment_text: None | str = None
+        self.update_text_body: None | bool = None
+
+    def set_tags(self, tags: Dict[str, str]) -> Self:
+        """
+        (Optional) List of tags to add to jira issue
+        Used to identify and update jira issues if one already
+        exists with the given title and tags. Defaults to {}.
+
+        Args:
+            tags (Dict[str, str]): Event tags
+        """
+        self.tags = tags
+        return self
+
+    def set_issue_type(self, issue_type: str) -> Self:
+        """
+        (Optional) Issue type for jira event, see
+        https://docs.datadoghq.com/api/latest/events/ for details.
+        Defaults to None.
+
+        Args:
+            issue_type (str): issue Type
+        """
+        self.issue_type = issue_type
+        return self
+
+    def set_fallback_comment_text(
```

Comment:
As you are adding these methods that are specific to the Jira Notifier, probably the idea of having one interface common to all notification channel will not fly.

I think about two options, you can pick:
1. Get rid of the common abstraction altogether and just expose multiple channels. Simpler.
2. Make the common interface only support notifications that are actually common, which can be: slack notification - datadog event - new jira ticket. No need to support adding comments.

Reply:
I discussed this with Ian yesterday, but I think the current inheritance format: Notifier, JiraNotifier(Notifier) is good for a few reasons: 
1. It makes setting fields and optional arguments less confusing for the user (compared to having everything being exposed), since some fields are only used in DD vs Jira. This way, it is also easier for users to know what is required and what is optional, and helps avoid potential missing fields.
2. The OOD makes creating a notification clearer and more easy to understand vs calling a single method that outputs to multiple channels

Reply:
> This way, it is also easier for users to know what is required and what is optional, and helps avoid potential missing fields.

But you may have noticed I advised you to remove the setters as well for mandatory fields.

Reply:
> It makes setting fields and optional arguments less confusing for the user (compared to having everything being exposed), since some fields are only used in DD vs Jira.

Could you please expand on this? I am not sure why things would be easier.

Reply:
> The OOD makes creating a notification clearer and more easy to understand vs calling a single method that outputs to multiple channels

How would you access the specific methods of the JiraNotifier that are not in the others ? The parent class can be useful if you have:
- common logic 
- you can actually use the same interface among the implementations IF you can reference a `Notifier` in the client code rather than a `JiraNotifier`. That is a requirement for polymorphism to be useful. Though You have some methods that are present only in the JiraNotifier. Thus the question on how you would use the Notifier.